solutions/silver/usaco-895.mdx

```diff
@@ -96,139 +113,160 @@ public class IcyPerimeter {
 <CPPSection>
 
 ```cpp
-// created by Tanish Tyagi
 #include <bits/stdc++.h>
-
 using namespace std;
-using ll = long long;
 
-const int MAXN = 1005;
-
-char grid[MAXN][MAXN];  // the grid itself
 int n;
-bool visited[MAXN][MAXN];  // keeps track of which nodes have been visited
-ll currA = 0;  // keeps track of the current area of the current blob of ice cream
-ll currP = 0;  // keeps track of the current perimeter of the blob of ice cream
-ll area = 0;
-ll perimeter = 0;
-
-void setIO(string name) {
-	ios_base::sync_with_stdio(0);
-	cin.tie(0);
-	if (len(name)) {
-		freopen((name + ".in").c_str(), "r", stdin);
-		freopen((name + ".out").c_str(), "w", stdout);
+vector<vector<char>> grid;
+vector<vector<bool>> visited;
+const vector<pair<int, int>> directions = {
+    {-1, 0}, {1, 0}, {0, -1}, {0, 1}};  // Movement directions (up, down, left, right)
+
+// BFS function to calculate the area and perimeter of a blob of ice cream
+pair<int, int> bfs(int x, int y) {
```

Comment:
hmm i personally would rather just put this in `main()` because making it its own function kind of forces u to have a bunch of global variables

Reply:
oh wait ic why u did it this way

hmm in that case it's fine, altho i would personally write a lambda